app/forms/LocationOfGoodsAddressFormProvider.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms
+
+import forms.mappings.Mappings
+import models.AddressLine._
+import models.{Address, CountryList}
+import play.api.data.Form
+import play.api.data.Forms.mapping
+import play.api.i18n.Messages
+
+import javax.inject.Inject
+
+class LocationOfGoodsAddressFormProvider @Inject() extends Mappings {
+
+  def apply(prefix: String, countryList: CountryList)(implicit messages: Messages): Form[Address] =
+    Form(
+      mapping(
+        AddressLine1.field -> {
+          lazy val args = Seq(AddressLine1.arg)
```

Comment:
Can probs remove lazy from the val(s) as they are always loaded. Lazy only loads when called therefore saving memory if not called. In this case we use args in all entities within the mapping.